### PR TITLE
Setup and run end to end

### PR DESCRIPTION
**Pull request by Lee-W**
The main change is documenting how to set the slack bot for ask-astro https://github.com/astronomer/ask-astro/blob/setup-and-run-end-to-end/docs/api/setup_slack_bot.md

Other are minor improvement and scripts to run ask-astro API locally

Closes: #32  #38

### CODE REVIEW

**Review by sunank200 — APPROVED** [review]
Over all looks good to me. Following needs to be added.
1. Installation of poetry and missing requirements if any.

File: .gitignore [hunk 1]
```diff
@@ -47,3 +47,179 @@ unittests.cfg
...
+
+# Jupyter Notebook
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
+#   For a library or package, you might want to ignore these files since the code is
```
[the remark this answers is not in the document]
Comment (Lee-W) [reply .gitignore:139]: These are generated through https://github.com/tj/git-extras/blob/master/man/git-ignore-io.md. I think it's more convenient to get commonly used gitignore for certain file types (in this case python)

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,57 @@
...
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
+
+
 [tool.black]
 line-length = 120
 target-version = ['py39']
 
-[tool.isort]
```
[the remark this answers is not in the document]
Comment (Lee-W) [reply pyproject.toml:5]: I thought it's included in ruff

File: tasks.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+@task
+def run_api_server_with_poetry(ctx: Context) -> None:
+    with ctx.cd("api"):
+        ctx.run("poetry run python -m ask_astro.app")
+
+
+@task
+def run_api_server(
```
[the remark this answers is not in the document]
Comment (Lee-W) [reply tasks.py:16]: sure, I was caught up in other tasks will come back and finish this afterward

[the remark this answers is not in the document]
Comment (Lee-W) [reply tasks.py]:
```python
>>> poetry run inv -l

Available tasks:

  init-api-server-poetry-env   Initialize the ask-astro API local poetry environment
  run-api-server-with-docker   Run ask-astro API server with docker
  run-api-server-with-poetry   Run ask-astro API server with poetry
  stop-api-server-container    Stop ask-astro API server container
```

```python
>>> poetry run inv run-api-server-with-docker -h

Usage: inv[oke] [--core-opts] run-api-server-with-docker [--options] [other tasks here ...]

Docstring:
  Run ask-astro API server with docker

Options:
  -b, --build-image                    build image before run API server
  -c STRING, --container-name=STRING   ask-astro API server container name
  -f, --[no-]follow-logs               follow logs after running container
  -i STRING, --image-name=STRING       ask-astro API server image name
```

[the remark this answers is not in the document]
Comment (Lee-W) [reply tasks.py]: Just updated. Now we'll have handy help text like the above

**Comment by Lee-W** [comment]
> Over all looks good to me. Following needs to be added.
> 
>     1. Installation of poetry and missing requirements if any.

could be improved but this is how it looks like now


image

**Review by pankajastro — APPROVED** [review]
Looks good, thanks!

**Pull request merged**